dist/ExtUtils-ParseXS/lib/perlxs.pod

```diff
@@ -1,2065 +1,4706 @@
+
+The C<PROTOTYPES> keyword corresponds to F<xsubpp>'s C<-prototypes> and
+C<-noprototypes> options.
+
+See L<perlsub/Prototypes> for more information about Perl prototypes.
+
+=head3 The EXPORT_XSUB_SYMBOLS: Keyword
+
+    EXPORT_XSUB_SYMBOLS: ENABLE | DISABLE
+
+This keyword is present since 5.16.0, and its value is disabled by
+default.
+
+Before 5.16.0, the C function which implemented an XSUB was exported.
+Since 5.16.0, it is declared C<static>. The old behaviour can be restored
+by enabling it. You are very unlikely to have a need for this keyword.
+
+=head3 The INCLUDE: Keyword
+
+    INCLUDE: const-xs.inc
+    INCLUDE: some_command |
+
+This keyword can be used to pull in the contents of another file to the
+"XS" part of an XS file. Unlike a top-level XS file, included files don't
+have a "C" first half, and the entire contents of the file are treated as
+XS, as if it had all been inserted at that line.
+
+One common use of C<INCLUDE> is to include constant definitions generated
+by F<ExtUtils::Constant>.
+
+If the parameters to the C<INCLUDE> keyword are followed by a pipe (C<|>)
+then the XS parser will interpret the parameters as a command. This
+feature is mildly deprecated in favour of the C<INCLUDE_COMMAND:>
+directive, as documented below. The latter can be used to ensure that the
+perl (if any) used in the command is the same as the one running the XS
+parser.
+
+=head3 The INCLUDE_COMMAND: Keyword
+
+    INCLUDE_COMMAND: $^X -e '...'
+
+Since 5.14.0.
+
+Similar to C<INCLUDE: some_command|> except that the C<|> is implicit, and
+it converts the special token C<$^X>, if present, to the path of the perl
+interpreter which is running the XS parser.
+
+=head3 The TYPEMAP: Keyword
+
+    TYPEMAP: <<EOF
+    myint  T_MYIV
+    INPUT
+        T_MYIV
+            $var = ($type)my_SvIV($arg)
+    OUTPUT
+        T_MYIV
+            my_sv_setiv($arg, (IV)$var);
+    EOF
+
+Since 5.16.0.
+
+Typemaps are mappings and code templates which allow the XS parser to
+automatically generate code snippets which convert between Perl and C
+values. The C<TYPEMAP> keyword can be used to embed typemap declarations
+directly into your XS code, instead of (or in addition to) typemaps in a
+separate file. Multiple such embedded typemaps will be processed in order
+of appearance in the XS code. Typemaps are processed in the order:
+
+=over
+
+=item *
+
+The system typemap file.
+
+=item *
+
+A local typemap file, typically specified by C<xsubpp -typemap typemap>
+in the F<Makefile>.
+
+=item *
+
+C<TYPEMAP:> entries, in order.
+
+=back
+
+The most recently-applied entries take precedence, so for example you can
+use C<TYPEMAP:> to individually override specific C<TYPEMAP>, C<INPUT>, or
+C<OUTPUT> entries in the system typemap. In general, typemap changes
+affect any subsequent XSUBs within the file, until further updates.  Note
+however that due a quirk in parsing, it is possible for a C<TYPEMAP:>
+entry immediately I<after> an XSUB to affect that XSUB.
```

Comment:
Wow! Can we provide more info here or a "how to avoid this"? Otherwise I wouldn't risk using it...